include/networkit/graph/Graph.hpp

```diff
@@ -590,25 +595,34 @@ class Graph final {
      */
     template <bool InEdges = false> class NeighborRange {
       public:
-        NeighborRange(const Graph &G, node u) : G(G), u(u){};
+        NeighborRange(const Graph &G, node u) : G(&G), u(u){
+			assert(exists[u]);
+        };
+
+        NeighborRange() : G(nullptr) {};
 
         NeighborIterator begin() const {
+            assert(!(G==nullptr));
```

Comment:
Should be either `assert(G)` or `assert(G != nullptr)`; I, personally, prefer the second.